1.5.1~rc4 build failure: assignment discards ‘const’ qualifier from pointer target type

On Ubuntu resolute (26.04), rc4 fails to build:

```
21:57:13 /build/reproducible-path/odyssey-1.5.1~rc4/sources/machinarium/tls.c:420:16: error: assignment discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
21:57:13   420 |         domain = strchr(name, '.');
21:57:13       |                ^
21:57:13 cc1: all warnings being treated as errors
```

Compiler there is gcc 15.2.0

https://jengus.postgresql.org/job/odyssey-binaries/17/architecture=amd64,distribution=resolute/console

This issue is still present in rc5.

```
13:43:29 /build/reproducible-path/odyssey-1.5.1~rc5/sources/machinarium/tls.c:420:16: error: assignment discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
13:43:29   420 |         domain = strchr(name, '.');
13:43:29       |                ^
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        os-version: ['jammy', 'noble', 'plucky', 'questing']
+        os-version: ['jammy', 'noble', 'plucky', 'questing', 'resolute']
         build-type: ['build_release']
     steps:
     - uses: actions/checkout@v6

--- a/sources/machinarium/tls.c
+++ b/sources/machinarium/tls.c
@@ -375,7 +375,7 @@ error:
 
 static inline int mm_tls_verify_name(char *cert_name, const char *name)
 {
-	char *cert_domain, *domain, *next_dot;
+	const char *cert_domain, *domain, *next_dot;
 	if (strcasecmp(cert_name, name) == 0) {
 		return 0;
 	}

